File: app/src/main/java/com/lzx/musiclib/imageloader/ActivityUtils.java
package com.lzx.musiclib.imageloader;

import android.app.Activity;
import android.content.Context;
import android.os.Build;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

/**
 * @author lzx
 * @date 2018/3/26
 */

public class ActivityUtils {

    public static boolean activityIsAlive(Context context) {
        if (context == null) {
            return false;
        }
        if (context instanceof Activity) {
            Activity activity = (Activity) context;
            return activityIsAlive(activity);
        } else {
            return true;
        }
    }

    public static boolean activityIsAlive(Activity activity) {
        if (activity == null) {
            return false;
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
            return !(activity.isDestroyed() || activity.isFinishing());
        } else {
            return !activity.isFinishing();
        }
    }

    public static void addFragmentToActivity(@NonNull FragmentManager fragmentManager,
                                             @NonNull Fragment fragment, int frameId) {
        checkNotNull(fragmentManager);
        checkNotNull(fragment);
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.add(frameId, fragment);
        transaction.commit();
    }

    public static <T> T checkNotNull(T reference) {
        if (reference == null) {
            throw new NullPointerException();
        } else {
            return reference;
        }
    }
}

File: app/src/main/java/com/lzx/musiclib/imageloader/ActivityUtils.kt
package com.lzx.musiclib.imageloader

import android.app.Activity
import android.content.Context
import android.os.Build
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager

/**
 * @author lzx
 * @date 2018/3/26
 */
object ActivityUtils {
    fun activityIsAlive(context: Context?): Boolean {
        if (context == null) {
            return false
        }
        return if (context is Activity) {
            activityIsAlive(context)
        } else {
            true
        }
    }

    fun activityIsAlive(activity: Activity?): Boolean {
        if (activity == null) {
            return false
        }
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
            !(activity.isDestroyed || activity.isFinishing)
        } else {
            !activity.isFinishing
        }
    }

    fun addFragmentToActivity(
        fragmentManager: FragmentManager,
        fragment: Fragment, frameId: Int
    ) {
        checkNotNull(fragmentManager)
        checkNotNull(fragment)
        val transaction =
            fragmentManager.beginTransaction()
        transaction.add(frameId, fragment)
        transaction.commit()
    }

    fun <T> checkNotNull(reference: T?): T {
        return if (reference == null) {
            throw NullPointerException()
        } else {
            reference
        }
    }
}